articles/azure-functions/functions-reference-powershell.md

```diff
@@ -0,0 +1,424 @@
+```
+
+At the root of the project, there's a shared [host.json](functions-host-json.md) file that can be used to configure the function app. Each function has a folder with its own code file (.ps1) and binding configuration file (function.json). The name of `function.json`'s parent directory is always the name of your function.
+
+The binding extensions required in [version 2.x](functions-versions.md) of the Functions runtime are defined in the `extensions.csproj` file, with the actual library files in the `bin` folder. When developing locally, you must [register binding extensions](functions-triggers-bindings.md#local-development-azure-functions-core-tools). When developing functions in the Azure portal, this registration is done for you.
+
+## Defining a PowerShell script to be a function
+
+By default, the Functions runtime looks for your function in `run.ps1`, where `run.ps1` shares the same parent directory as its corresponding `function.json`.
+
+Your script is passed a number of arguments on execution. The powershell script takes in parameters that match the names of all the input bindings. In addition to those inputs, a parameter is available to you called `TriggerMetadata` that contains additional information on the trigger that started the function. You can define this at the top of your script by adding a `param` block:
+
+```powershell
+# $TriggerMetadata is optional here. If you don't need it, you can safely remove it from the param block
+param($MyFirstInputBinding, $MySecondInputBinding, $TriggerMetadata)
+```
+
+## Bindings
+
+In PowerShell, [bindings](functions-triggers-bindings.md) are configured and defined in a function's function.json. Functions interact with bindings a number of ways.
+
+### Reading trigger and input data
+
+Trigger and input bindings (bindings that have `direction` set to `in`) can be read as parameters passed to your function. The `name` property defined in `function.json` will be the name of the parameter, in the `param` block. Since PowerShell uses a parameter binder, the order of the parameters doesn't matter. However, it's a best practice to follow the order of the bindings defined in the `function.json`.
+
+```powershell
+param($MyFirstInputBinding, $MySecondInputBinding)
+```
+
+### Writing data
+
+Outputs (bindings that have `direction` set to `out`) can be written to by a function using a cmdlet called `Push-OutputBinding` that is available within the Functions runtime. In all cases, the `name` property of the binding as defined in `function.json` corresponds to the `Name` parameter of the `Push-OutputBinding` cmdlet.
+
+```powershell
+param($MyFirstInputBinding, $MySecondInputBinding)
+
+Push-OutputBinding -Name myQueue -Value $myValue
+
+# or
+
+Push-OutputBinding myQueue $myValue
+```
+
+You can also pass in a hashtable that contains the names of output bindings to their values.
+
+```powershell
+param($MyFirstInputBinding, $MySecondInputBinding)
+
+$hashtable @{
+    myQueue = "myData"
```

Comment:
To be consistent with the first example above, may be better to have this as:

```suggestion
    myQueue = $myValue
```

Reply:
then the script doesn't make sense because `$myValue` is not defined. Worth it for consistency?

Reply:
fixed